app/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,24 @@
+class SessionsController < ApplicationController
+  before_action :require_logged_out, only: [:new]
+
+  # TODO: message for require_logged_out should be suppressed
+
+  def create
+    user = User.find_by(email: params[:email])
+
+    if user && user.authenticate(params[:password])
+      flash[:notice] = "Welcome, #{user.full_name}"
+      session[:user_id] = user.id
+      redirect_to home_path
+    else
+      flash[:error] = "Invalid email or password"
```

Comment:
When we render, we want to use `flash.now` rather than just `flash`, because otherwise the flash message carries to next GET request as well - and that creates bad user experience.